### DISCUSSION TITLE
Set Locale

### DISCUSSION THREAD

**Discussion by hugoj-goncalves** [community/community#26491] [Actions]
<p>I couldn’t find any topics about changing the runner locale.</p>
<p>I have a few tests that breaks because of the decimal separator, so I’m trying to change the locale of the runner before running my build process.</p>
<pre><code>- name: SetLocale
        run: |
          sudo apt-get update &amp;&amp; sudo apt-get install tzdata locales -y &amp;&amp; sudo locale-gen pt_BR.UTF-8
          sudo localectl set-locale LANG="pt_BR.UTF-8"
          export LANG="pt_BR.UTF-8"
          sudo update-locale
          echo "Testing language settings"
          echo "All languages..."
          locale -a
          echo "Actual locale"
          locale
          echo "Actual numeric settings"
          locale -c -k LC_NUMERIC
          echo "LANG Variable"
          echo "$LANG"
          echo "LocaleCtl"
          localectl status
          echo "Printing decimal"
          printf "%'.3f\n" 123123,123
</code></pre>
<p>and the output:</p>
<pre><code>Generating locales (this might take a while)...
  pt_BR.UTF-8... done
Generation complete.
**Testing language settings All languages...** C
C.UTF-8
en_US.utf8
POSIX
pt_BR.utf8
**Actual locale** LANG=pt_BR.UTF-8
LANGUAGE=
LC_CTYPE="pt_BR.UTF-8"
LC_NUMERIC="pt_BR.UTF-8"
LC_TIME="pt_BR.UTF-8"
LC_COLLATE="pt_BR.UTF-8"
LC_MONETARY="pt_BR.UTF-8"
LC_MESSAGES="pt_BR.UTF-8"
LC_PAPER="pt_BR.UTF-8"
LC_NAME="pt_BR.UTF-8"
LC_ADDRESS="pt_BR.UTF-8"
LC_TELEPHONE="pt_BR.UTF-8"
LC_MEASUREMENT="pt_BR.UTF-8"
LC_IDENTIFICATION="pt_BR.UTF-8"
LC_ALL=
**Actual numeric settings** LC_NUMERIC
decimal_point=","
thousands_sep="."
grouping=3;3
numeric-decimal-point-wc=44
numeric-thousands-sep-wc=46
numeric-codeset="UTF-8"
**LANG Variable** pt_BR.UTF-8
**LocaleCtl** System Locale: LANG=pt_BR.UTF-8
       VC Keymap: n/a
      X11 Layout: us
       X11 Model: pc105
**Printing decimal** &lt;font color="#FF0000"&gt;/home/runner/work/_temp/912d399d-17df-48b4-a4d0-b8001d971635.sh: line 18: printf: 123123,123: invalid number&lt;/font&gt;0.000
&lt;font color="#FF0000"&gt;##[error]Process completed with exit code 1.&lt;/font&gt;
</code></pre>
<p>I have no idea what I’m missing here…</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/hugoj-goncalves">@hugoj-goncalves</a> ,</p>
<p>In the <strong>printf</strong> command, it can’t recognize a long integer number argument that segmented with " <strong>,</strong>". If you want to input a long integer number, just do not segment it with " <strong>,</strong>".<br>
So the following command line can be OK:</p>
<pre><code>printf "%'.3f\n" 123123123
</code></pre>
<p>the output is: </p>
<pre><code>123123123.000
</code></pre>

**Comment by hugoj-goncalves** [comment]
<p><img src="https://user-images.githubusercontent.com/10927715/181098012-21893b7d-6984-4984-add0-e8ab753e6e9f.png" alt="printf.png" data-base62-sha1="rGWO0hVCT20qe4UWE9hrBYSn1jO" width="260" height="39"> <img src="https://user-images.githubusercontent.com/10927715/181098013-2d568a1c-0c75-4879-9284-a62c34cdec13.png" alt="printf2.png" data-base62-sha1="fsxh4qg4cXSSPBVpBx9Q5mGRFoP" width="273" height="36"></p>
<p>that should be totally locale dependant.</p>
<p>I need it to be formated this way tough.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/hugoj-goncalves">@hugoj-goncalves</a> ,</p>
<p>I also tested on my local machines, and also got the " <strong>invalid number</strong>" error.<br>
I tested and study according to some articles about the <strong>printf</strong> command, I think the <strong>number formats</strong> in the system settings is available to the <strong>output</strong> of the <strong>printf</strong> command, but NOT available to the <strong>input</strong> arguments. I searched and did not find any specific option for setting the <strong>number formats</strong> of the <strong>input</strong> on the <strong>printf</strong> command.</p>

**Comment by hugoj-goncalves** [comment]
<p>It does work if you have the right number settings. If you look at the output from <br>
<img src="https://user-images.githubusercontent.com/10927715/181098018-20872903-6e42-4d0c-ac74-c874980d1051.png" alt="printf.png" data-base62-sha1="iJnCQCFOjbjuWHgmecQQkb1dofB" width="416" height="38"></p>
<pre><code>locale -c -k LC_NUMERIC

**Actual numeric settings** LC_NUMERIC
decimal_point=","
thousands_sep="."
grouping=3;3
numeric-decimal-point-wc=44
numeric-thousands-sep-wc=46
numeric-codeset="UTF-8"
</code></pre>
<p> it should respect these settings, for input and output<br>
but that’s not the actual issue for me… let’s just forget about the printf for now.<br>
the whole <strong>locale</strong> settings are being <strong>ignored</strong> , even tough the timezone worked just fine.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/338a72d61f2bbb457d9b0125aa98ac74c517b275.png" data-download-href="/uploads/short-url/7lWYuCi4SdZxgigUzsLMhwR2DOZ.png?dl=1" title="timezone.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/10927715/181098019-27b37fd5-14b9-40b4-8f8b-2b8ced73801b.png" alt="timezone.png" data-base62-sha1="7lWYuCi4SdZxgigUzsLMhwR2DOZ" width="511" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/338a72d61f2bbb457d9b0125aa98ac74c517b275_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">timezone.png</span><span class="informations">538×526 37.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
<strong>date</strong> should output something like this:
<pre><code>seg 20 abr 2020 10:49:13 -03</code></pre>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/hugoj-goncalves">@hugoj-goncalves</a> ,</p>
<p>I have helped you create a new a new issue ticket (<a href="https://github.com/actions/runner/issues/439" rel="nofollow noopener"><strong>actions/runner #439</strong></a>) to report this question to the appropriate engineering team for further investigation and evaluation.</p>
<p>You can follow the issue ticket and add your comments here.</p>

**Comment by hugoj-goncalves** [comment]
<p>thank you very much, didn’t know where to create the issue :wink:</p>
